src/core/metrics.py

```diff
@@ -1338,6 +1502,11 @@ def get_cluster_gpu_info() -> Dict[str, Any]:
         "temperatures": [],
         "power_usage": [],
     }
+    
+    nvidia_count = 0
+    intel_count = 0
+    
+    # Try NVIDIA DCGM metrics
```

Comment:
Since the logic for Nvidia metrics is similar to Intel Gaudi metrics, what about create a common function to reduce code repetition?